### DISCUSSION TITLE
PhotoPrism Geo Location in 64 bit

### DISCUSSION THREAD

**Discussion by keif888** [photoprism/photoprism#4526] [Show and tell]
I was in the process of creating a Pull request when I saw that @lastzero had just checked the equivalent in...

As I am unsure how to reach out I thought I would put it here.
My branch is here.  https://github.com/keif888/photoprism/tree/GeoLocation-64bit
It has an additional acceptance test https://github.com/keif888/photoprism/blob/GeoLocation-64bit/frontend/tests/acceptance/acceptance-public/photos-geo64.js which checks that the UI renders the results out to 7 decimal places without rounding.  This test fails on the version before today, 
![image](https://github.com/user-attachments/assets/4e4d65e4-67b7-4466-9504-0a4189ba2352)
but should work against @lastzero latest version with 64 bit float location fields, as it worked against mine.
![image](https://github.com/user-attachments/assets/aa075706-d9a3-40ca-a22b-4506fcefbbe6)

I also have redone the unit tests to check out to 6 decimal places.
https://github.com/keif888/photoprism/blob/GeoLocation-64bit/internal/entity/photo_location_test.go
And all the EXIF tests, where I recomputed the DMS values to decimals.
https://github.com/keif888/photoprism/blob/GeoLocation-64bit/internal/meta/json_motion_test.go

This is how I am testing the float values using InDelta.
```
		assert.InDelta(t, 35.778152, data.Lat, 0.0000009)
		assert.InDelta(t, -78.636875, data.Lng, 0.0000009)
```

**Comment by lastzero** [comment]
Excellent, thanks for letting us know! We'd love to invite you to our Contributors Chat if you reach out via Matrix/Element or email to hello@photoprism.app :+1:

**Reply by lastzero** [reply]
As for your suggestions: I agree that the tests would benefit from improvements, such as using `assert.InDelta()` instead of `assert.InEpsilon()` and checking the actual precision of values (depending on where they come from).

From what I remember, we have had situations in the past where floating point numbers were slightly different depending on e.g. the CPU architecture or Exiftool version. So our tests are usually not too strict about their exact value or precision.
